app/src/main/java/org/mifos/mobile/utils/Network.kt

```diff
@@ -15,9 +15,13 @@ object Network {
      * @param context
      * @return
      */
-    private fun getNetworkInfo(context: Context): NetworkInfo {
-        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
-        return cm.activeNetworkInfo
+    private fun getNetworkInfo(context: Context): NetworkInfo? {
```

Comment:
Are these changes needed?

Reply:
yes as previously function required non null active network info but if not connected to internet it will return null and that was error.

Whole function can be optimised as

private fun getNetworkInfo(context: Context): NetworkInfo? {
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        return cm.activeNetworkInfo
    }
I will change them asap.

Reply:
Sorry for those commits, initially i did only one but while working on other issue it got commited here automatically. I have recently started using git so i was unaware of such issue. Is there anything I can do now to merge all commits?